### PR TITLE
Remove arbitrary code execution in byte size settings

### PR DESCRIPTION
**Pull request by quantum5**
Settings like `TOTAL_MEMORY` could be used for arbitrary code execution through carefully crafted values of command line flags. This may be problematic in say, online services similar to https://godbolt.org, if they happen to use Emscripten.

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -1077,16 +1077,19 @@ def check(value):
...
 
 
+SIZE_SUFFIXES = {suffix: 1024 ** i for i, suffix in enumerate(['b', 'kb', 'mb', 'gb', 'tb'])}
+
+
 def expand_byte_size_suffixes(value):
-  """Given a string with arithmetic and/or KB/MB size suffixes, such as
-  "1024*1024" or "32MB", computes how many bytes that is and returns it as an
-  integer.
   """
```
Comment (sbc100) [inline tools/shared.py]: Docstrings normally start on the same line as the opening triple quote. See https://www.python.org/dev/peps/pep-0257/

**Comment by sbc100** [comment]
For what its worth I think tools like gotbolt already have to assume they can't trust the compiler and therefore run compilers in secure sandboxes.   Se while I don't think the gotbolt argument is good enough reason to make this change I still support this simplification

File: tools/shared.py [hunk 2]
```diff
@@ -1077,16 +1077,21 @@ def check(value):
...
-  integer.
+  """Given a string with KB/MB size suffixes, such as "32MB", computes how
+  many bytes that is and returns it as an integer.
   """
-  value = value.lower().replace('tb', '*1024*1024*1024*1024').replace('gb', '*1024*1024*1024').replace('mb', '*1024*1024').replace('kb', '*1024').replace('b', '')
-  try:
-    return eval(value)
-  except Exception:
-    raise Exception("Invalid byte size, valid suffixes: KB, MB, GB, TB")
+  match = re.match(r'\s*(\d+)\s*([kmgt]?b)$', value, re.I)
```
Comment (tlively) [inline tools/shared.py]: Does this still match both lowercase and uppercase prefixes?

Comment (quantum5) [reply tools/shared.py]: Yes, that's what `re.I` is for.

**Pull request merged**